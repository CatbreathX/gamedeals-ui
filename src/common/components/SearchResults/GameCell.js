import PropTypes from 'prop-types';

export const GameCell = ({
  dealId,
  title,
  normalPrice,
  storeName,
  salePrice,
  thumbnailImage,
  priceSaving,
  metacriticRating,
  steamRatingText,
  dealRating,
}) => (
  <div key={dealId}>
    <div>
      <img src={thumbnailImage} alt={title} />
    </div>

    <div>
      <span className="game_cell_title">{title}</span>
      <span className="game_cell_store">{storeName}</span>
    </div>

    <div>
      <span className="game_cell_normal-price">{normalPrice}</span>
      <span className="game_cell_sale-price">{salePrice}</span>
      <span className="game_cell_price-saving">{priceSaving}</span>
    </div>

    <div>
      <span className="game_cell_metacritic-rating">{metacriticRating}</span>
      <span className="game_cell_steam-rating">{steamRatingText}</span>
      <span className="game_cell_deal-rating">{dealRating}</span>
    </div>

  </div>
);

GameCell.propTypes = {
  dealId: PropTypes.string.isRequired,
  dealRating: PropTypes.string.isRequired,
  metacriticRating: PropTypes.string.isRequired,
  normalPrice: PropTypes.string.isRequired,
  priceSaving: PropTypes.string.isRequired,
  salePrice: PropTypes.string.isRequired,
  steamRatingText: PropTypes.string.isRequired,
  storeName: PropTypes.string.isRequired,
  thumbnailImage: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};
