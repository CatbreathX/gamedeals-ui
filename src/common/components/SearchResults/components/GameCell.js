import {
  DataItem,
  DataCell,
  DataValue, DataCellContainer,
} from 'common/components/SearchResults/components/layout/styles';
import PropTypes from 'prop-types';

export const GameCell = ({
  dealId,
  title,
  normalPrice,
  storeName,
  salePrice,
  thumbnailImage,
  priceSaving,
  metacriticRating,
  steamRatingText,
  dealRating,
}) => (
  <DataCellContainer key={dealId}>
    <DataCell>
      <img src={thumbnailImage} alt={title} />
    </DataCell>

    <DataCell>
      <DataItem className="game_cell_title">{title}</DataItem>
      <DataItem className="game_cell_store">{storeName}</DataItem>
    </DataCell>

    <DataCell>
      <DataItem>Normal Price</DataItem>
      <DataValue>{normalPrice}</DataValue>
    </DataCell>

    <DataCell>
      <DataItem>Sale Price</DataItem>
      <DataValue>{salePrice}</DataValue>
    </DataCell>

    <DataCell>
      <DataItem>Savings</DataItem>
      <DataValue>{priceSaving}</DataValue>
    </DataCell>

    <DataCell>
      <DataItem className="game_cell_metacritic-rating">{metacriticRating}</DataItem>
      <DataItem className="game_cell_steam-rating">{steamRatingText}</DataItem>
      <DataItem className="game_cell_deal-rating">{dealRating}</DataItem>
    </DataCell>

  </DataCellContainer>
);

GameCell.propTypes = {
  dealId: PropTypes.string.isRequired,
  dealRating: PropTypes.string.isRequired,
  metacriticRating: PropTypes.string.isRequired,
  normalPrice: PropTypes.string.isRequired,
  priceSaving: PropTypes.string.isRequired,
  salePrice: PropTypes.string.isRequired,
  steamRatingText: PropTypes.string.isRequired,
  storeName: PropTypes.string.isRequired,
  thumbnailImage: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};
