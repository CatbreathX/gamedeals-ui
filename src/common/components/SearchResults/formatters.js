export const formatString = (value) => value;

export const formatMetaCriticScore = (value) => (value === '0' ? '' : value);
