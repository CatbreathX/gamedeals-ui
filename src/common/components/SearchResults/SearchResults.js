import { formatString } from 'common/components/SearchResults/formatters';
import {
  CircularProgressSpinner,
  ProgressContainer,
  Table,
  TableHeadingTr,
  Td,
  Th,
  Tr,
} from 'common/components/SearchResults/layout/styles';
import { camelCase } from 'lodash/string';
import PropTypes from 'prop-types';

const noop = () => {
};

export const SearchResults = ({
  columns,
  dataset,
  isError,
  isLoading,
  isSuccess,
  dataRowClicked,
  headerClicked,
  pagination,
}) => {
  const onHeaderClicked = (header, i) => {
    const normalizedHeader = camelCase(header);
    headerClicked(normalizedHeader, i);
  };

  return (
    <>
      {isLoading &&
        <ProgressContainer>
          <CircularProgressSpinner />
        </ProgressContainer>
      }

      {pagination}
      <Table>
        <thead>
          <TableHeadingTr>
            {columns.map((column, i) => (
              <Th
                key={column.header}
                onClick={() => {
                  onHeaderClicked(column.header, i);
                }}
              >
                {column.header}
              </Th>
            ))}
          </TableHeadingTr>
        </thead>
        <tbody>
          {isError &&
          <tr>
            <td colSpan={columns.length} className="search-results__error">
              An Error Occurred getting
              your data!
            </td>
          </tr>
          }

          {isSuccess && dataset.length === 0 &&
          renderNoResults(columns)
          }
          {isSuccess && dataset.length > 0 &&
          dataset.map((row, rowIndex) => renderRow(columns, row, rowIndex, dataRowClicked))
          }
        </tbody>
      </Table>
      {!isLoading && pagination}
    </>
  );
};

SearchResults.propTypes = {
  dataset: PropTypes.instanceOf(Array),
  isLoading: PropTypes.bool.isRequired,
  isError: PropTypes.bool.isRequired,
  isSuccess: PropTypes.bool.isRequired,
  headerClicked: PropTypes.func,
  dataRowClicked: PropTypes.func,
  columns: PropTypes.arrayOf(
    PropTypes.shape({
      header: PropTypes.string.isRequired,
      accessor: PropTypes.string.isRequired,
      className: PropTypes.string,
      headerRenderer: PropTypes.func,
      valueFormatter: PropTypes.func,
    }),
  ).isRequired,
  pagination: PropTypes.node,
};

SearchResults.defaultProps = {
  dataset: [],
  headerClicked: noop(),
  dataRowClicked: noop(),
  pagination: null,
};

const renderNoResults = (columns) => (
  <tr>
    <td colSpan={columns.length}>
      No Results Found
    </td>
  </tr>
);

const renderRow = (columns, row, rowIndex, dataRowClicked) => (
  <Tr
    key={`row-${rowIndex}`}
    onClick={() => dataRowClicked(row)}
  >
    {renderColumns(columns, row, rowIndex)}
  </Tr>
);

const renderColumns = (columns, rowData, rowIndex) => columns.map((column, columnIndex) => {
  const value = rowData[column.accessor];
  const valueFormatter = column.valueFormatter || formatString;
  const formattedValue = valueFormatter(value, rowData);
  const key = `column-${value}-${rowIndex}-${columnIndex}`;

  return (
    <Td key={key}>
      {formattedValue}
    </Td>
  );
});
