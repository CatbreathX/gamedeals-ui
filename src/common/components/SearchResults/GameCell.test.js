import { GameCell } from 'common/components/SearchResults/GameCell';
import React from 'react';
import { renderComponent } from 'unit/componentRenderers';

describe('GameCell', () => {
  test('should render correctly', () => {
    const component = renderComponent(
      <GameCell  title="Elite Dangerous" price="10.00" />,
    );

    expect(component).toMatchSnapshot();
  });
});
