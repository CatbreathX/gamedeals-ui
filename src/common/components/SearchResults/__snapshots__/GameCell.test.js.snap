// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameCell should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <img
            alt="Elite Dangerous"
            src="https://BASE_IMAGE/game/1/image"
          />
        </div>
        <div>
          <span
            class="game_cell_title"
          >
            Elite Dangerous
          </span>
          <span
            class="game_cell_store"
          >
            Steam
          </span>
        </div>
        <div>
          <span
            class="game_cell_normal-price"
          >
            29.99
          </span>
          <span
            class="game_cell_sale-price"
          >
            5.99
          </span>
          <span
            class="game_cell_price-saving"
          >
            24.00
          </span>
        </div>
        <div>
          <span
            class="game_cell_metacritic-rating"
          >
            85
          </span>
          <span
            class="game_cell_steam-rating"
          >
            Mostly Positive
          </span>
          <span
            class="game_cell_deal-rating"
          >
            4.0
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <img
          alt="Elite Dangerous"
          src="https://BASE_IMAGE/game/1/image"
        />
      </div>
      <div>
        <span
          class="game_cell_title"
        >
          Elite Dangerous
        </span>
        <span
          class="game_cell_store"
        >
          Steam
        </span>
      </div>
      <div>
        <span
          class="game_cell_normal-price"
        >
          29.99
        </span>
        <span
          class="game_cell_sale-price"
        >
          5.99
        </span>
        <span
          class="game_cell_price-saving"
        >
          24.00
        </span>
      </div>
      <div>
        <span
          class="game_cell_metacritic-rating"
        >
          85
        </span>
        <span
          class="game_cell_steam-rating"
        >
          Mostly Positive
        </span>
        <span
          class="game_cell_deal-rating"
        >
          4.0
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
